packages/kit/src/runtime/client/router.js

```diff
@@ -185,7 +185,10 @@ export class Router {
 			if (a instanceof SVGAElement ? a.target.baseVal : a.target) return;
 
 			// Check if new url only differs by hash
-			if (url.href.split('#')[0] === location.href.split('#')[0]) {
+			if (
+				url.href.split('#').length > 1 &&
+				url.href.split('#')[0] === location.href.split('#')[0]
+			) {
```

Comment:
We should try to not repeat splitting more than once

```suggestion
			const [base, hash] = url.href.split('#');
			if (hash && base === location.href.split('#')[0]) {
```